### PR TITLE
fix: selected asset on buy/swap

### PR DESCRIPTION
**Pull request by dgacevic**
https://bayes.atlassian.net/browse/RWPD-466

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift [hunk 1]
```diff
@@ -39,13 +39,23 @@ class SwapInteractor: NSObject, Interactor, SwapViewActions {
...
                 let enabled = self?.dataStore?.currencies.filter { cur in currencies.map { $0.name }.contains(cur.code) }
                 
-                guard let from = enabled?.first,
+                var fromCurrency: Currency?
+                
+                if let from = self?.dataStore?.from?.currency {
+                    fromCurrency = from
+                } else if let from = enabled?.first {
+                    fromCurrency = from
+                }
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift:48]: Can be a one liner

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift [hunk 2]
```diff
@@ -39,13 +39,23 @@ class SwapInteractor: NSObject, Interactor, SwapViewActions {
...
+                
+                guard let from = fromCurrency,
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift:50]: let fromCurrency,

File: brd-ios/breadwallet/src/Redesign/Core/CLEANerSwift/VIP/Coordinator.swift [hunk 3]
```diff
@@ -114,6 +116,8 @@ class BaseCoordinator: NSObject, Coordinatable {
                     vc?.dataStore?.paymentMethod = type
                     vc?.dataStore?.coreSystem = coreSystem
                     vc?.dataStore?.keyStore = keyStore
+                    guard let currency = selectedCurrency else { return }
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Core/CLEANerSwift/VIP/Coordinator.swift:119]: do you actually need to guard statement?

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/Main/BuyStore.swift [hunk 4]
```diff
@@ -30,6 +30,7 @@ class BuyStore: NSObject, BaseDataStore, BuyDataStore {
     var values: BuyModels.Amounts.ViewAction = .init()
     var paymentMethod: PaymentCard.PaymentType? {
         didSet {
+            guard toAmount == nil else { return }
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Buy/Main/BuyStore.swift:33]: is this ok? why is this needed?

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift [hunk 5]
```diff
@@ -39,14 +39,18 @@ class SwapInteractor: NSObject, Interactor, SwapViewActions {
...
+                
+                guard let fromCurrency,
+                      let toCurrency = enabled?.first(where: { $0.code != fromCurrency.code })
                 else {
                     self?.presenter?.presentError(actionResponse: .init(error: ExchangeErrors.selectAssets))
                     return
                 }
-                self?.dataStore?.from = .zero(from)
-                self?.dataStore?.to = .zero(to)
+                if self?.dataStore?.from == nil {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift:50]: why check nil?

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/Main/BuyStore.swift (hunk 4 above)
Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Scenes/Buy/Main/BuyStore.swift:33]: the code after this part is setting the toAmount value for the screen (selected crypto). If i provide that from previous screen I dont want this code to execute, because it will override selected crypto with default one (atm BTC)

File: brd-ios/breadwallet/src/Redesign/Core/CLEANerSwift/VIP/Coordinator.swift (hunk 3 above)
Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Core/CLEANerSwift/VIP/Coordinator.swift:119]: If i dont unwrap here i will have to unwrap in the next line, which would be a lot uglier

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift (hunk 5 above)
Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Scenes/Swap/Main/SwapInteractor.swift:50]: there's no point in setting datastore.from value if it's already set

**Review by kenanmamedoff — APPROVED** [review]
